lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
+      this.tokenProvider.addTokens(UserCredentialSecurityTokenProvider.serializeToken(token));
```

Comment:
My concern is if this code path is used/tested somewhere. Because from the current code base, it is not used as default is false. 

Given this is a pretty isolated PL and the usage is in another PL, then I am fine for the merge.